core/src/test/java/org/apache/accumulo/core/clientImpl/lexicoder/ByteUtilsTest.java

```diff
@@ -78,8 +83,9 @@ public void testEscape() {
     byte[] escaped = ByteUtils.escape(bytes);
     assertArrayEquals(bytes, ByteUtils.unescape(escaped));
 
-    // these bytes would cause an ArrayIndexOutOfBounds in the past
-    byte[] errorBytes = {0x01, 0x01, 0x01};
+    // incomplete bytes would cause an ArrayIndexOutOfBounds in the past
+    exception.expect(IllegalArgumentException.class);
```

Comment:
If the code immediately after this throws an exception, then the rest of the test will not be executed.

Reply:
+1 to that comment. exception expectations should be declared immediately prior to the invocation that triggers the exception, and a call to `fail()` should be used immediately after that invocation.

Reply:
Good catch.  I will have to fix where I have used this incorrectly in the past.

Reply:
> call to `fail()` should be used immediately after that invocation.

Isn't this redundant since the test will fail if it doesn't throw the ExpectedException?

Reply:
No. Using `fail()` gives you the opportunity to provide a more useful error message (it takes a String as an arg), and it allows you to fence off the code, so that somebody doesn't unintentionally add additional lines, which alter the test scenario.

I would only consider it redundant in relatively trivial test cases, where the line where the expected failure occurs is obvious (for example: single line test cases).

Reply:
Fair enough.  It seemed redundant here since it was just checking for the exception.